sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -918,18 +968,34 @@ class RocksDBSuite extends AlsoTestWithRocksDBFeatures with SharedSparkSession
       }
       // Check that snapshots and changelogs get purged correctly.
       db.doMaintenance()
-      assert(snapshotVersionsPresent(remoteDir) === Seq(30, 60))
+
+      if (enableStateStoreCheckpointIds && colFamiliesEnabled) {
```

Comment:
nit: isn't it duplicated with the next if statement?

Reply:
No - one is checking for snapshot version and the other for changelog version